components/eamxx/src/share/io/scream_scorpio_types.hpp

```diff
@@ -0,0 +1,154 @@
+#ifndef SCREAM_SCORPIO_TYPES_HPP
+#define SCREAM_SCORPIO_TYPES_HPP
+
+#include <string>
+#include <map>
+#include <vector>
+#include <memory>
+
+namespace scream {
+namespace scorpio {
+
+// Enum denoting how we open a file
+enum FileMode {
+  Unset = 0,
+  Read = 1,
+  Write = 2,
+  Append = Read | Write
+};
+std::string e2str (const FileMode mode);
+
+// I/O types supported
+enum IOType {
+  // Default I/O type is used to let the code choose I/O type as needed (via CIME)
+  DefaultIOType = 0,
+  NetCDF,
+  PnetCDF,
+  Adios,
+  Hdf5,
+  Invalid
+};
+
+IOType str2iotype(const std::string &str);
+std::string iotype2str(const IOType iotype);
+
+// The type used by PIOc for offsets
+using offset_t = std::int64_t;
+
+/*
+ * The following PIOxyz types each represent an entity that
+ * is associated in scorpio with an id. For each of them, we add some
+ * additional metadata, to avoid having to call PIOc_inq_xyz every time
+ * we need such information.
+ * While these types are defined in this public header, the customers
+ * of scream_io will never be able to get any object of these types out
+ * of the internal database. In particular, all data is stored in a
+ * ScorpioSession singleton class, whose declaration is hidden inside
+ * scream_scorpio_interface.cpp.
+ */
+
+// The basic common data of any PIO entity
+struct PIOEntity {
+  int ncid = -1;      // PIO access all data via their id
+  std::string name;   // In EAMxx, we prefer to use a string name than a number id
+};
+
+// An entity that is linked to a specific file
+// This allows an entity to retrieve its file
+struct PIOFileEntity : public PIOEntity {
+  int fid = -1;       // Allow retrieving file from the entity
+};
+
+// struct PIOAtt : public PIOFileEntity {
```

Comment:
any reason this is commented out?

Reply:
Yeah, I was thinking of adding a PIOAtt ministruct, to allow to easy query if a certain att existed (much like for PIOVar/PIODim). But I realized that for atts there's really not much metadata, so it's prob not worth it. I will delete these lines when I'm ready to clean it up.